.gitignore

```diff
@@ -39,6 +39,10 @@
 /.idea
 /cmake-build-debug
 /x64/Debug
+/out
+
+# Visual Studio Specific CMake Settings
```

Comment:
It is not Visual Studio specific instead it is used by cmake all the times.
and replace `CMakeSettings.json` with `/CMakeSettings.json`